annotation/src/main/java/io/smallrye/common/annotation/NonBlocking.java

```diff
@@ -0,0 +1,26 @@
+package io.smallrye.common.annotation;
+
+import static java.lang.annotation.ElementType.*;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation used to indicate that the annotated method is inherently non-blocking and so can be
+ * executed on a <em>non-blockable</em> thread (I/O threads, event loops...) without the need to offload the work to
+ * another thread. If the caller thread can be blocked, it should also be safe to execute the method on that thread.
+ *
+ * It's up to the framework relying on this annotation do define the exact behavior, like <em>what</em> thread is
+ * considered as a <em>non-blockable</em> thread.
+ *
```

Comment:
You probably want to separate the paragraphs with `<p>`, not just empty line.

Reply:
Good catch!